src/encounter.c

```diff
@@ -54,982 +55,951 @@
+
     case 3:
-        sub_02050B1C(v1->unk_10, fieldSystem);
+        UpdateFieldSystemFromDTO(encounter->dto, fieldSystem);
 
-        if (v1->unk_10->resultMask == BATTLE_RESULT_CAPTURED_MON) {
-            TVBroadcast *v4 = SaveData_TVBroadcast(fieldSystem->saveData);
-            Pokemon *v5 = Party_GetPokemonBySlotIndex(v1->unk_10->parties[1], 0);
+        if (encounter->dto->resultMask == BATTLE_RESULT_CAPTURED_MON) {
+            TVBroadcast *broadcast = SaveData_TVBroadcast(fieldSystem->saveData);
+            Pokemon *caughtMon = Party_GetPokemonBySlotIndex(encounter->dto->parties[1], 0);
 
-            sub_0206D018(v4, v5);
+            sub_0206D018(broadcast, caughtMon);
         }
 
-        sub_020518B0(fieldSystem, v1->unk_10);
-
-        if (((*v3) == 0) && (v1->unk_10->resultMask != BATTLE_RESULT_CAPTURED_MON)) {
-            Location *v6;
+        UpdateGameRecords(fieldSystem, encounter->dto);
 
-            v6 = FieldOverworldState_GetSpecialLocation(SaveData_GetFieldOverworldState(fieldSystem->saveData));
-            FieldTask_ChangeMapByLocation(taskMan, v6);
+        if (*ballCount == 0 && encounter->dto->resultMask != BATTLE_RESULT_CAPTURED_MON) {
+            Location *location = FieldOverworldState_GetSpecialLocation(SaveData_GetFieldOverworldState(fieldSystem->saveData));
+            FieldTask_ChangeMapByLocation(task, location);
         }
 
-        sub_02051988(fieldSystem, v1->unk_10);
-        (*v2)++;
+        UpdateJournal(fieldSystem, encounter->dto);
+        (*state)++;
         break;
+
     case 4:
-        FieldTask_StartFieldMap(taskMan);
-        (*v2)++;
+        FieldTransition_StartMap(task);
+        (*state)++;
         break;
+
     case 5:
         MapObjectMan_UnpauseAllMovement(fieldSystem->mapObjMan);
-        sub_020558F0(taskMan);
-        (*v2)++;
+        FieldTransition_FadeIn(task);
+        (*state)++;
         break;
+
     case 6:
-        if ((*v3) == 0) {
-            if (v1->unk_10->resultMask == BATTLE_RESULT_CAPTURED_MON) {
-                ScriptManager_Start(taskMan, 8802, NULL, NULL);
+        if (*ballCount == 0) {
+            if (encounter->dto->resultMask == BATTLE_RESULT_CAPTURED_MON) {
+                ScriptManager_Start(task, 8802, NULL, NULL);
             } else {
-                ScriptManager_Start(taskMan, 8809, NULL, NULL);
+                ScriptManager_Start(task, 8809, NULL, NULL);
             }
         } else {
-            PCBoxes *v7 = SaveData_PCBoxes(fieldSystem->saveData);
-            Party *v8 = Party_GetFromSavedata(fieldSystem->saveData);
+            PCBoxes *boxes = SaveData_PCBoxes(fieldSystem->saveData);
+            Party *party = Party_GetFromSavedata(fieldSystem->saveData);
 
-            if ((PCBoxes_FirstEmptyBox(v7) == 18) && (Party_GetCurrentCount(v8) == 6)) {
-                ScriptManager_Start(taskMan, 8822, NULL, NULL);
+            if (PCBoxes_FirstEmptyBox(boxes) == 18 && Party_GetCurrentCount(party) == MAX_PARTY_SIZE) {
```

Comment:
**suggestion**: Might not hurt to go ahead and add a `MAX_PC_BOXES` constant or something to `include/unk_020797C8.h`, so there's less stuff like this to hunt down later.